h2o-algos/src/test/java/hex/glm/GLMBasicTestMultinomial.java

```diff
@@ -49,18 +50,77 @@ public static void cleanUp() {
 
 
   @Test
+  public void testCovtypeNoIntercept(){
+    GLMParameters params = new GLMParameters(Family.multinomial);
+    GLMModel model = null;
+    Frame preds = null;
+    Vec weights = _covtype.anyVec().makeCon(1);
+    Key k = Key.<Frame>make("cov_with_weights");
+    Frame f = new Frame(k,_covtype.names(),_covtype.vecs());
+    f.add("weights",weights);
+    DKV.put(f);
+    try {
+      params._response_column = "C55";
+      params._train = k;
+      params._valid = _covtype._key;
+      params._objective_epsilon = 1e-6;
+      params._beta_epsilon = 1e-4;
+      params._weights_column = "weights";
+      params._missing_values_handling = DeepLearningModel.DeepLearningParameters.MissingValuesHandling.Skip;
+      params._intercept = false;
+      double[] alpha = new double[]{0,.5,.1};
+      Solver s = Solver.L_BFGS;
+      System.out.println("solver = " + s);
+      params._solver = s;
+      params._max_iterations = 5000;
+      for (int i = 0; i < alpha.length; ++i) {
+        params._alpha = new double[]{alpha[i]};
+//        params._lambda[0] = lambda[i];
+        model = new GLM(params).trainModel().get();
+        System.out.println(model.coefficients());
+//        Assert.assertEquals(0,model.coefficients().get("Intercept"),0);
+        double [][] bs = model._output.getNormBetaMultinomial();
+        for(double [] b:bs)
+          Assert.assertEquals(0,b[b.length-1],0);
+        System.out.println(model._output._model_summary);
+        System.out.println(model._output._training_metrics);
+        System.out.println(model._output._validation_metrics);
+        preds = model.score(_covtype);
+        ModelMetricsMultinomialGLM mmTrain = (ModelMetricsMultinomialGLM) hex.ModelMetricsMultinomial.getFromDKV(model, _covtype);
+        assertTrue(model._output._training_metrics.equals(mmTrain));
+        model.delete();
+        model = null;
+        preds.delete();
+        preds = null;
+      }
+    } finally{
+      weights.remove();
+      DKV.remove(k);
+      if(model != null)model.delete();
+      if(preds != null)preds.delete();
+    }
+  }
+
+  @Test
   public void testCovtypeBasic(){
     GLMParameters params = new GLMParameters(Family.multinomial);
     GLMModel model = null;
     Frame preds = null;
+    Vec weights = _covtype.anyVec().makeCon(1);
+    Key k = Key.<Frame>make("cov_with_weights");
+    Frame f = new Frame(k,_covtype.names(),_covtype.vecs());
+    f.add("weights",weights);
+    DKV.put(f);
     try {
       params._response_column = "C55";
-      params._train = _covtype._key;
+      params._train = k;
       params._valid = _covtype._key;
       params._lambda = new double[]{4.881e-05};
       params._alpha = new double[]{1};
       params._objective_epsilon = 1e-6;
       params._beta_epsilon = 1e-4;
+      params._weights_column = "weights";
+      params._missing_values_handling = DeepLearningModel.DeepLearningParameters.MissingValuesHandling.Skip;
```

Comment:
That's where MissingValueHandling is defined. Probably should be moved to shared library.